mpdev/internal/apply/resource.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apply
+
+import (
+	"strings"
+)
+
+// Resource represents a KRM resource that can be applied
+type Resource interface {
+	Apply(registry *registry) error
+	GetReference() Reference
+	GetDependencies() []Reference
+}
+
+// Reference allows a Resource to reference another Resource as part of its
+// specification. The combination of Group, Kind, Name MUST be unique for all
+// applied resources.
+type Reference struct {
+	Group string
+	Kind  string
+	Name  string
+}
+
+// ReferenceMap is a mapping between KRM references and the resource object.
+type ReferenceMap map[Reference]Resource
```

Comment:
rm?

Reply:
Changing to `refMap`, rm seems to short to me